zone/client_packet.cpp

```diff
@@ -11794,578 +11853,658 @@ void Client::Handle_OP_RaidCommand(const EQApplicationPacket *app)
-								if (raid->members[x].GroupNumber == group) {
-									raid->SetGroupLeader(raid_command_packet->leader_name, false);
-									raid->SetGroupLeader(raid->members[x].membername);
-									raid->UpdateGroupAAs(group);
-									break;
-								}
-							}
-						}
+		break;
+	}
+	case RaidCommandDisband: {
+		Raid* raid = entity_list.GetRaidByClient(this);
+		Client* c_to_disband = entity_list.GetClientByName(raid_command_packet->leader_name);
+		Client* c_doing_disband = entity_list.GetClientByName(raid_command_packet->player_name);
 
+		if (raid) {
+			raid->VerifyRaid();
+			uint32 group = raid->GetGroup(raid_command_packet->leader_name);
+
+			if (RuleB(Bots, Enabled)) {
+				Bot* const b_to_disband = entity_list.GetBotByBotName(raid_command_packet->leader_name);
+
+				//Added to remove all bots if the Bot_Owner is removed from the Raid
+				//Does not camp the Bots, just removes from the raid
+				if (c_to_disband) {
+					raid->HandleBotGroupDisband(c_to_disband->CharacterID());
+					raid->RemoveMember(raid_command_packet->leader_name);
+					if (raid->IsLeader(c_to_disband->GetName())) {
+						uint32 i = raid->GetPlayerIndex(raid_command_packet->leader_name);
+						raid->SetNewRaidLeader(i);
 					}
-					if (raid->members[i].IsRaidLeader) {
-						for (int x = 0; x < MAX_RAID_MEMBERS; x++) {
-							if (strlen(raid->members[x].membername) > 0 && strcmp(raid->members[x].membername, raid->members[i].membername) != 0)
-							{
-								raid->SetRaidLeader(raid->members[i].membername, raid->members[x].membername);
-								raid->UpdateRaidAAs();
-								raid->SendAllRaidLeadershipAA();
+					raid->SendGroupDisband(c_to_disband);
+					raid->GroupUpdate(group);
+					if (!raid->RaidCount() || !raid->GetLeader()) {
+						raid->DisbandRaid();
+					}
+					break;
+				} else if (b_to_disband) {
+					uint32 gid = raid->GetGroup(b_to_disband->GetName());
+
+					if (gid < 12 && (raid->IsGroupLeader(b_to_disband->GetName()) || raid->GroupCount(gid) < 2)) {
+						uint32 owner_id = b_to_disband->CastToBot()->GetOwner()->CastToClient()->CharacterID();
+						raid->HandleBotGroupDisband(owner_id, gid);
+
+					} else if (b_to_disband && raid->IsRaidMember(b_to_disband->GetName())) {
+						Bot::RemoveBotFromRaid(b_to_disband);
+
+					} else if (gid < 12 && raid->GetGroupLeader(gid) && raid->GetGroupLeader(gid)->IsBot()) {
+						c_doing_disband->Message(
+							Chat::Yellow,
+							fmt::format(
+								"{} is in a Bot Group.  Please disband {} instead to remove the entire Bot group.",
+								raid_command_packet->leader_name,
+								raid->GetGroupLeader(gid)->CastToBot()->GetName()
+							).c_str()
+						);
+					}
+
+					break;
+				}
+			}
+			if (group < 12) {
+				uint32 i = raid->GetPlayerIndex(raid_command_packet->leader_name);
+				if (raid->members[i].IsGroupLeader) { //assign group leader to someone else
+					for (int x = 0; x < MAX_RAID_MEMBERS; x++) {
```

Comment:
You can range through the array versus doing the integer iteration

eg

```
for (auto & m : members) {
```

Accessing

```cpp
m.member->DoThing()
```